### DISCUSSION TITLE
OpenModellica: How to use variables in Block-Diagramm view & how to get own models to be loaded

### DISCUSSION THREAD

**Discussion by Land-Vogt** [OpenModelica/OpenModelica#11505] [Q&A]
Hello,

In order not to post too much, I'll put my two questions into one discussion.

1. values for parameters (e.g. spring stiffness or initial length) can be specified in the block diagram view.  Is it possible to fill in these fields with variable names and specify them in a central location (e.g. in a block provided for this purpose, in the code view or in a text file)? If so, what would a simple example look like or where could I find something about this?

2) I have built my own sub-model that I use in other models. This works very well, the simulations run without any problems. Only when I close OpenModellica and open it again, the sub-model is not loaded. The sub-model then has to be dragged in again and linked to the connectors. Can this be solved e.g. by a kind of path definition or similar?

I would be grateful for any advice!

Best regards!

**Comment by casella** [comment]
Regarding 1., the simplest way is demonstrated in this attached package: [ExampleRecord.zip](https://github.com/OpenModelica/OpenModelica/files/13266835/ExampleRecord.zip). You can drag and drop a record on a model, and then refer to it in components. The records that you drag and drop have their default values, but you can also change them locally in the model.

Regarding 2., my suggestion is to put everything in a single package, so you don't have to load the different models and sub-models separately.
